File: 0011-container-with-most-water/0011-container-with-most-water.java
class Solution {
    public int maxArea(int[] height) {
        int l=0;
        int r=height.length-1;
        int min=0,max=0,area=0;
        while(l<r){
            min=Math.min(height[l],height[r]);
            area=min*(r-l);
            if(height[l]<height[r]){
                int x=l;
                while(l<r&&height[x]>=height[l]) {
                    l++;
                }
            }
            else{
                int x=r;
                while(l<r&&height[x]>=height[r]) {
                    r--;
                }
            }
            max=Math.max(max,area);
        }
        return max;
    }
}

File: 0011-container-with-most-water/0011-container-with-most-water.cpp
class Solution {
public:
    int maxArea(vector<int>& height) {
        int maxwater = 0;
        int left  = 0;
        int right = height.size()-1;
        while(left < right){
            maxwater = max(maxwater,(min(height[left],height[right])*(right - left)));
            if(height[left] > height[right]) right--;
            else left++;
        }
        return maxwater;
    }
};